roombasim/pittras/task/takeoff_task.py

```diff
@@ -0,0 +1,47 @@
+
+import numpy as np
+
+import roombasim.config as cfg
+
+from roombasim.ai import Task
+
+class TakeoffTaskState:
+    init = 0
+    request_arm = 1
+    pause_before_takeoff = 2
+    takeoff = 3
+    ascend = 4
+    ascend_angle = 5
+    done = 6
+    failed = 7
+
+class TakeoffTask(Task):
+
+    def __init__(self):
+        # null velocities
+        self.zero_xy_vel = np.array([0,0], dtype=np.float64)
+
+        # state machine
+        self._state = TakeoffTaskState.init
+
+    def update(self, delta, elapsed, state_controller, environment):
+        drone_state = state_controller.query('DroneState', environment)
+
+        height = drone_state['z_pos']
+        control_z_vel = drone_state['z_vel']
+
+        # Pause before ramping up the motors
+        if elapsed > cfg.PITTRAS_DELAY_BEFORE_TAKEOFF * 1000:
```

Comment:
At the moment, `elapsed` corresponds to the amount of time since the start of the entire mission. Although, it would probably make more sense for tasks to receive the elapsed time since the task was created.

Reply:
That would make sense. I could store the initial value of `elapsed` in an instance variable and base this calculation on that value.

Though it would make more sense to give the task the elapsed time since it was switched to.

This task is likely only going to be called once per run of the mission. While for other tasks, they may be switched to multiple times.

Reply:
Yeah, since this would probably be the first task scheduled, I'll go ahead and accept the PR as it works under that circumstance. Then once we switch over to giving tasks elapsed time since their creation it should be a seamless transition. I also realized there is no way to signal that a task has terminated so I'll try to build up more of a framework to get that to work.